### PR TITLE
[Rust] Implement new APIs for Vm and ImportObj

### LINKED ISSUE #601: [Rust SDK] Support loading wasm bytecode from an array
The C API allows the developer to load a WASM module from a memory buffer.

```
WasmEdge_VMLoadWasmFromBuffer
```

The Rust SDK should support this as well.

### LINKED ISSUE #750: [Rust SDK] WasmEdge_ImportObjectCreateWasmEdgeProcess

### LINKED ISSUE #751: [Rust SDK] WasmEdge_ImportObjectInitWasmEdgeProcess

### LINKED ISSUE #773: [Rust SDK] WasmEdge_VMRegisterModuleFromFile

### LINKED ISSUE #774: [Rust SDK] WasmEdge_VMRegisterModuleFromBuffer

### LINKED ISSUE #775: [Rust SDK] WasmEdge_VMRegisterModuleFromASTModule

### LINKED ISSUE #777: [Rust SDK] WasmEdge_VMLoadWasmFromBuffer

### LINKED ISSUE #800: [Rust SDK] WasmEdge_VMGetFunctionList

### LINKED ISSUE #801: [Rust SDK] WasmEdge_VMExecuteRegistered

### PR DESCRIPTION
**Pull request by apepkuss**
The following C-APIs are involved in this PR:
- VM
  - Fix #773 
  - Fix #774 
  - Fix #775 
  - Fix #777 
  - Fix #601 
  - Fix #801 
  - Fix #800 
- ImportObj
  - WasmEdge_ImportObjectWASIGetExitCode
  - Fix #750 
  - Fix #751

### CODE REVIEW

File: bindings/rust/wasmedge-sys/src/types.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use super::wasmedge;
 pub type WasmEdgeProposal = wasmedge::WasmEdge_Proposal;
-pub type HostRegistration = wasmedge::WasmEdge_HostRegistration;
+// pub type HostRegistration = wasmedge::WasmEdge_HostRegistration;
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/types.rs:3]: Why do you comment out this line?

**Review by hydai — CHANGES_REQUESTED** [review]

File: bindings/rust/wasmedge-sys/src/types.rs (hunk 1 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/types.rs:3]: I'll remove the code line. Thanks!

**Review by hydai — DISMISSED** [review]

**Comment by hydai** [comment]
Looks like there is a conflict in `types.rs`. @apepkuss Please resolve it and I am going to merge this PR. Thanks.

**Comment by apepkuss** [comment]
@hydai I'll resolve the conflicts, thanks a lot!

**Comment by apepkuss** [comment]
Hi @hydai, the conflicts have been resolved, please review the changes again, thanks a lot!

**Review by hydai — APPROVED** [review]

**Comment by hydai** [comment]
Looks like GitHub broken again...
This PR is merged: https://github.com/WasmEdge/WasmEdge/commit/ebbb3684fb4932935b4f99c6cab7d4195c2d5897
However, it doesn't show this got merged. I am closing this manually.

**Pull request closed without merging**